src/App.tsx

```diff
@@ -26,70 +26,128 @@ type ReorderOptions = {
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+export class App extends Component<{}, State> {
+  state = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  componentDidMount = () => {
+    const buttons = document.querySelector('.buttons');
+
+    buttons?.addEventListener('click', ({ target }) => {
+      if (target) {
+        const button = target as HTMLButtonElement;
+
+        if (buttons) {
+          Array.from(buttons.children).forEach(el => {
+            el.classList.add('is-light');
+          });
+        }
+
+        if (button.matches('.button')) {
+          button.classList.remove('is-light');
+        }
+      }
+    });
+  };
+
+  render() {
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className="button is-info is-light"
+            onClick={() => {
+              this.setState({ sortType: SortType.ALPHABET });
+            }}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className="button is-success is-light"
+            onClick={() => {
+              this.setState({ sortType: SortType.LENGTH });
+            }}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className="button is-warning is-light"
+            onClick={() => {
+              this.setState((state) => ({ isReversed: !state.isReversed }));
+            }}
+          >
+            Reverse
+          </button>
+
+          {(this.state.sortType !== SortType.NONE || this.state.isReversed)
+          && (
+            <button
+              type="button"
+              className="button is-danger is-light"
+              onClick={() => {
+                this.setState({ isReversed: false, sortType: SortType.NONE });
+              }}
+            >
+              Reset
+            </button>
+          )}
+        </div>
 
-      <ul>
         <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
+          <ul>
+            {getReorderedGoods(goodsFromServer, this.state).map(good => {
+              return (
+                <li data-cy="Good" key={Date.now() + Math.random()}>{good}</li>
+              );
```

Comment:
Call getReorderedGoods befor return in render method.

Do not use such values as keys Date.now() + Math.random() it may cause rerender each time